gprofiler/utils/__init__.py

```diff
@@ -147,9 +151,13 @@ def start_process(
             cmd = [f"{staticx_dir}/.staticx.interp", "--library-path", staticx_dir] + cmd
         else:
             env = env if env is not None else os.environ.copy()
-            # ensure `TMPDIR` env is propagated to the child processes (used by staticx)
-            if "TMPDIR" not in env and "TMPDIR" in os.environ:
+            if tmpdir is not None:
+                tmpdir.mkdir(exist_ok=True)
+                env["TMPDIR"] = tmpdir.as_posix()
+            elif "TMPDIR" not in env and "TMPDIR" in os.environ:
+                # ensure `TMPDIR` env is propagated to the child processes (used by staticx)
```

Comment:
I’m not sure this is the right approach
maybe should we use `tmpdir` always relative to the current env[“TMPDIR”] (e.g. `f”{env[“TMPDIR”]}/{tmpdir}”`)

Reply:
Is there a practical difference? Doesn't the current flow always create the tmpdir passed to this function under $TMPDIR?

Reply:
Ok, but the tmpdir arg, whenever it is passed it is always `Path(os.environ["TMPDIR"]) / ("pyperf_" + random_prefix())` which is under $TMPDIR. So there is no difference because with both approaches the tmpdir is created under $TMPDIR.

Reply:
there’s no practical differences… I raised this to suggest what’s the better approach